src/fs_io.h

```diff
@@ -30,16 +30,26 @@ class dread_ctrl
     void *client_data;
 };
 
-// POD
 class dwrite_q
 {
+    MEMPROXY_CLASS(dwrite_q);
 public:
-    off_t file_offset;
-    char *buf;
-    size_t len;
-    size_t buf_offset;
-    dwrite_q *next;
-    FREE *free_func;
+    dwrite_q() = default;
+    dwrite_q(const size_t, char * = nullptr, FREE * = nullptr);
```

Comment:
To remove the risk of memory-leaking two-argument calls, please split this constructor into two:

```suggestion
    dwrite_q(size_t);
    dwrite_q(size_t, char *, FREE *);
```

The above suggestion also removes the unwanted/unused "const" from the first parameter declaration.

If possible, the first constructor should call the second one to reduce code duplication.

Reply:
This violates @rousskov principles, "always use const" - wherein we are required to always use "const" keyword even if the type is implicitly copied-to-stack and/or the original is automatically treated as constant.